heim-net/src/sys/windows/nic.rs

```diff
@@ -1,57 +1,284 @@
+
+    Some(Address::Inet(
+        SocketAddrV4::new(ip4, port)
+    ))
+}
+
+fn sockaddr_to_ipv6(sa: SOCKET_ADDRESS) -> Option<Address> {
+    // Check this sockaddr can fit a SOCKADDR_IN6 (two shorts, two longs, and a 16-byte struct)
+    // (see https://docs.microsoft.com/en-us/windows/win32/winsock/sockaddr-2)
+    if (sa.iSockaddrLength as usize) < std::mem::size_of::<SOCKADDR_IN6>() {
+        return None;
+    }
+
+    let p_sa6 = sa.lpSockaddr as *const SOCKADDR_IN6;
+    if p_sa6.is_null() {
+        return None;
+    }
+    let sa6 = unsafe{ *p_sa6 };
+
+    let ip6_data = unsafe{ sa6.sin6_addr.u.Byte() };
+    let ip6 = Ipv6Addr::from(*ip6_data);
+    let port = sa6.sin6_port;
+    let flow_info = sa6.sin6_flowinfo;
+    let scope_id = unsafe{ *sa6.u.sin6_scope_id() };
+
+    Some(Address::Inet6(
+        SocketAddrV6::new(ip6, port, flow_info, scope_id )
+    ))
+}
+
+/// Generate an IPv4 netmask from a prefix length (Rust equivalent of ConvertLengthToIpv4Mask())
+fn ipv4_netmask_from(length: u8) -> Ipv4Addr {
+    let mask = match length {
+        len if len <= 32 => u32::max_value().checked_shl(32 - len as u32).unwrap_or(0),
+        _ /* invalid value */ => u32::max_value(),
+    };
+    Ipv4Addr::from(mask)
+}
+
+/// Generate an IPv6 netmask from a prefix length
+fn ipv6_netmask_from(length: u8) -> Ipv6Addr {
+    let mask = match length {
+        len if len <= 128 => u128::max_value().checked_shl(128 - len as u32).unwrap_or(0),
+        _ /* invalid value */ => u128::max_value(),
+    };
+    Ipv6Addr::from(mask)
+}
+
+fn ipv4_netmask_address_from(length: u8) -> Address {
+    Address::Inet(SocketAddrV4::new(ipv4_netmask_from(length), 0))
+}
+fn ipv6_netmask_address_from(length: u8) -> Address {
+    Address::Inet6(SocketAddrV6::new(ipv6_netmask_from(length), 0, 0, 0))
+}
+
+
+
+
 
 impl Nic {
     pub fn name(&self) -> &str {
-        unimplemented!()
+        &self.friendly_name
     }
 
-    pub fn address(&self) -> Address {
-        unimplemented!()
+    pub fn index(&self) -> Option<u32> {
+        Some(self.index)
     }
 
-    pub fn netmask(&self) -> Option<Address> {
-        unimplemented!()
+    pub fn guid(&self) -> &str {
+        &self.guid
+    }
+
+    pub fn address(&self) -> Address {
+        self.address.unwrap_or_else(||
+            Address::Inet(SocketAddrV4::new(Ipv4Addr::new(0,0,0,0), 0))
+        )
     }
 
-    pub fn broadcast(&self) -> Option<Address> {
-        unimplemented!()
+    pub fn netmask(&self) -> Option<Address> {
+        self.netmask
     }
 
     pub fn destination(&self) -> Option<Address> {
-        unimplemented!()
+        // TODO: we could implement something one day
+        None
     }
 
     pub fn is_up(&self) -> bool {
-        unimplemented!()
+        // TODO: not sure how to tell on Windows
+        true
     }
 
     pub fn is_running(&self) -> bool {
-        unimplemented!()
+        // TODO: not sure how to tell on Windows
+        true
```

Comment:
As a reminder: after all reviews it would be nice to create tracking issues for this and above `TODO`s in order not to forget about these things.

Reply:
https://todo.jasonet.co/ I've used this in some projects to great effect. It's a bot that automatically opens issues for TODO comments when merging a PR. Thought you might be interested ^^.